ui-modules/catalog/app/views/catalog/catalog.template.html

```diff
@@ -98,18 +98,29 @@ <h4 class="media-heading">
                     </div>
 
                     <div class="extra" uib-collapse="isCollapsed">
+                        <!-- note entities are not filtered as search changes; but they will take an initial search; not intuitive -->
                         <div ng-init="filteredEntities = (bundle.types | bundleTypeFilter:'org.apache.brooklyn.api.entity.Entity' | filter:state.search)" ng-show="::filteredEntities.length > 0">
                             <ul class="list-group list-group-types">
                                 <li class="list-group-item typed-group-table-header">
                                     <ng-pluralize count="::filteredEntities.length"
                                               when="{'one': 'Entity', 'other': 'Entities'}">
                                     </ng-pluralize>
                                 </li>
-                                <li ng-repeat="entity in filteredEntities track by (entity.containingBundle + entity.symbolicName + entity.version)"
+                                <li ng-repeat="entity in ::filteredEntities track by (entity.containingBundle + entity.symbolicName + entity.version)"
+                                        ng-if="!isCollapsed && $index < pagination.itemsPerPage"
                                         ng-class="{'deprecated': entity.deprecated, 'disabled': entity.disabled}"
                                         class="list-group-item">
-                                    <type-item bundle="bundle" type="entity"></type-item>
+                                    <type-item bundle="::bundle" type="::entity"></type-item>
```

Comment:
the `::` here doesn't achieve much, as the digest re-renders `type-item` isolated scope anyway :(